incubator/hnc/pkg/stats/metrics.go

```diff
@@ -2,22 +2,45 @@ package stats
 
 import (
 	"context"
+	"math"
+	"time"
 
 	ocstats "go.opencensus.io/stats"
 	ocview "go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 var (
-	hierConfigReconcileTotal = ocstats.Int64("hierconfig_reconcile_total", "The number of HierConfig reconciliations happened", "times")
-	hierConfigWritesTotal    = ocstats.Int64("hierconfig_writes_total", "The number of HierConfig writes happened during HierConfig reconciliations", "times")
+	hierConfigReconcileTotal      = ocstats.Int64("hierconfig_reconcile_total", "The total number of HierConfig reconciliations happened", "reconciliations")
+	hierConfigReconcileConcurrent = ocstats.Int64("hierconfig_reconcile_concurrent_peak", "The peak concurrent HierConfig reconciliations happened in the last reporting period", "reconciliations")
+	hierConfigWritesTotal         = ocstats.Int64("hierconfig_writes_total", "The number of HierConfig writes happened during HierConfig reconciliations", "writes")
+	namespaceWritesTotal          = ocstats.Int64("namespace_writes_total", "The number of namespace writes happened during HierConfig reconciliations", "writes")
+	objectReconcileTotal          = ocstats.Int64("object_reconcile_total", "The total number of object reconciliations happened", "reconciliations")
+	objectReconcileConcurrent     = ocstats.Int64("object_reconcile_concurrent_peak", "The peak concurrent object reconciliations happened in the last reporting period", "reconciliations")
+	objectWritesTotal             = ocstats.Int64("object_writes_total", "The number of object writes happened during object reconciliations", "writes")
 )
 
 var (
-	hierReconcileView = &ocview.View{
+	maxConcurrentHierConfigReconcile int64
+	maxConcurrentObjectReconcile     map[schema.GroupKind]int64
+)
+
+var KeyGK, _ = tag.NewKey("GK")
```

Comment:
Doc comment?

Reply:
Also maybe `KeyGroupKind` (if that's what is is) would be nice? I don't feel strongly though, up to you.